addons/survey/static/src/js/survey_form.js

```diff
@@ -129,14 +137,16 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend({
     },
 
     _onMatrixBtnClick: function (event) {
-        if (!this.options.readonly) {
-            var $target = $(event.currentTarget);
-            var $input = $target.find('input');
-            if ($input.attr('type') === 'radio') {
-                $input.prop("checked", true).trigger('change');
-            } else {
-                $input.prop("checked", !$input.prop("checked")).trigger('change');
-            }
+        if (this.readonly) {
```

Comment:
why not only modify the existing condition in `if (!this.readonly):` ?
I know you love preconditions but here, I don't really see the advantages

Reply:
Because I love preconditions.
Do I need another excuse? 👼